clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10711,9 +10716,12 @@ static void getTripleBasedSYCLPostLinkOpts(const ToolChain &TC,
     bool SplitEsimd = TCArgs.hasFlag(
         options::OPT_fsycl_device_code_split_esimd,
         options::OPT_fno_sycl_device_code_split_esimd, SplitEsimdByDefault);
-    // Symbol file and specialization constant info generation is mandatory -
+    bool IsUsingLTO = TC.getDriver().isUsingLTO(/*IsDeviceOffloadAction=*/true);
```

Comment:
We will need to add this logic inside clang-linker-wrapper as well. Please see comment in Line 10671. I can help with it if need be.

Thanks

Reply:
I haven't added any thinLTO changes to clang-linker-wrapper yet. I will do that in my next change and will definitely include this. Thanks.